src/jsil/jsil_parser.h

```diff
@@ -16,27 +16,29 @@ Author: Michael Tautschnig, tautschn@amazon.com
 
 #include "jsil_parse_tree.h"
 
-class jsil_parsert;
-int yyjsilparse(jsil_parsert &);
-void jsil_scanner_init(jsil_parsert &);
-
 class jsil_parsert:public parsert
 {
 public:
   explicit jsil_parsert(message_handlert &message_handler)
     : parsert(message_handler)
   {
+    // not thread-safe, simplistic check that we don't attempt to do reentrant
```

Comment:
I'd drop the reference to threads, as misleading -- recursive usage also won't work. Simply say that the parser has global state, and is hence not reentrant.

Reply:
Reworded.